state/feature.go

```diff
@@ -26,6 +26,8 @@ const (
 	FeatureQueryAPI Feature = "QUERY_API"
 	// FeatureTTL is the feature that supports TTLs.
 	FeatureTTL Feature = "TTL"
+	// FeaturePartitionKey is the feature that supports the partition
+	FeaturePartitionKey Feature = "PARTITION_KEY"
```

Comment:
Please remove from here. We do not want to introduce this as a feature for all state stores because it doesn't apply to all state stores. You don't have to (and should not) formally define this feature.